spec.bs

```diff
@@ -248,6 +251,92 @@ If using a centralized mechanism for fetching keys, user agents should have a re
 
 When using a registration process, it is recommended that user agents apply an expiration date to registration requests, to allow for the removal of deprecated or no longer active issuers.
 
+VOPRF Methods {#voprf-methods}
+====================================
+
+This document encodes protocol messages in TLS notation from [[!RFC8446]].
+
+For Issue (corresponding to **BlindEvaluate** function of the [[!VOPRF]] protocol), the Request and Response are serialized as:
+
+```
+// Scalars are elliptic curve scalars of length Ns (determined by the curve, 48 for P-384).
+// ECPoints are elliptic curve points encoded using the X9.62 Uncompressed point representation (determined by the curve, 97 for P-384).
+
+struct {
+  uint16 count;
+  ECPoint nonces[count];
+} IssueRequest;
+
+struct {
+  ECPoint blinded;
+  ECPoint evaluated;
+} SignedNonce;
+
+struct {
+  Scalar c;
+  Scalar s;
+} DLEQProof;
+
+struct {
+  uint16 issued;
+  uint32 key_id = keyID;
+  SignedNonce signed[issued];
+  opaque proof<1..2^16-1>; // Bytestring containing a serialized DLEQProof struct.
+} IssueResponse;
```

Comment:
For definitions like this, as a first approximation I'd put a `<dfn>` for them just before the code block, and then use [=IssueResponse=] to link to it. There might be a better way to do that if we think longer...